gems/pending/disk/modules/MSVSDiffDisk.rb

```diff
@@ -29,38 +29,12 @@ def d_init
       @locators << MSCommon::PARENT_LOCATOR.decode(MSCommon.header["parent_loc#{idx}"])
       next if @locators[idx - 1]['platform_code'] == "\000\000\000\000"
       locator = @locators[idx - 1]
-      case locator['platform_code']
-      when "Wi2r"
-      # Deprecated (no information on format)
-
-      when "Wi2k"
-      # Deprecated (no information on format)
-
-      when "W2ru"
-      # Relative path. Would much rather have absolute path.
-      # NOTE: Absolute path always accompanies relative path.
-      # getParentPathWin(locator)
-
-      when "W2ku"
-        # Absolute path - this is the one.
+      if locator['platform_code'] == "W2ku"
         getParentPathWin(locator)
         getParent(locator)
-
-      when "Mac "
-      #
-      # TODO: need details on Mac Alias Blob.
-      #
-      # getParentPathMac(locator)
-
-      when "MacX"
-        # Is platform spanning even something we should do?
-        # NOTE: Oleg says don't worry about it for the present (03/14/2007).
-        getParentPathMacX(locator)
-        getParent(locator)
-
       end
-      # raise "No compatible parent locator found" if @parent == nil
     end
+    raise "No compatible parent locator found" if @parent == nil
```

Comment:
Sorry but, since we're raising within this loop where we didn't previously, do we need to cleanup/close/disconnect `@ms_disk_file` and `@ @hyperv_connection` in the case of this exception?